python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -90,19 +100,88 @@ def __new__(
-
     def with_replaced_attrs(self, **kwargs: Any) -> "RunRequest":
         fields = self._asdict()
         for k in fields.keys():
             if k in kwargs:
                 fields[k] = kwargs[k]
         return RunRequest(**fields)
 
+    def with_resolved_tags_and_config(
+        self,
+        target_definition: Union["JobDefinition", "UnresolvedAssetJobDefinition"],
+        current_time: Optional[datetime] = None,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
+    ) -> "RunRequest":
+        from dagster._core.definitions.job_definition import JobDefinition
+
+        if self.partition_key is None:
+            check.failed(
+                "Cannot resolve partition for run request without partition key",
+            )
+
+        partitions_def = target_definition.partitions_def
+        if partitions_def is None:
+            check.failed(
+                "Cannot resolve partition for run request without partitions_def",
+            )
+        partitions_def = cast(PartitionsDefinition, partitions_def)
+
+        partitioned_config = (
+            target_definition.partitioned_config
+            if isinstance(target_definition, JobDefinition)
+            else PartitionedConfig.from_flexible_config(target_definition.config, partitions_def)
+        )
+        if partitioned_config is None:
+            check.failed(
+                "Cannot resolve partition for run request on unpartitioned job",
+            )
+
+        # Relies on the partitions def to throw an error if the partition does not exist
+        partition = partitions_def.get_partition(
+            self.partition_key,
+            current_time=current_time,
+            dynamic_partitions_store=dynamic_partitions_store,
+        )
+
+        get_run_request_tags = lambda partition: (
+            {
+                **self.tags,
+                **partitioned_config.get_tags_for_partition_key(
+                    partition,
+                    dynamic_partitions_store=dynamic_partitions_store,
+                    current_time=current_time,
+                    job_name=target_definition.name,
+                ),
+            }
+            if self.tags
+            else partitioned_config.get_tags_for_partition_key(
+                partition,
+                dynamic_partitions_store=dynamic_partitions_store,
+                current_time=current_time,
+                job_name=target_definition.name,
+            )
+        )
+
+        return self.with_replaced_attrs(
+            run_config=self.run_config
+            if self.run_config
+            else partitioned_config.get_run_config_for_partition_key(
+                partition.name,
+                dynamic_partitions_store=dynamic_partitions_store,
+                current_time=current_time,
+            ),
+            tags=get_run_request_tags(partition.name),
+        )
+
+    def has_resolved_partition_run_request(self) -> bool:
```

Comment:
This name makes it sound like "this RunRequest contains a(nother) RunRequest with a resolved partition". `has_resolved_partition` would be clearer I think.